waspc/data/Generator/templates/server/src/jobs/pgBossJob.js

```diff
@@ -0,0 +1,37 @@
+import { boss } from './pgBoss.js'
+import { SubmittedJob } from './SubmittedJob.js'
+
+class PgBossSubmittedJob extends SubmittedJob {
+  constructor(jobName, jobId, executor, pgBoss) {
+    super(jobName, jobId, executor)
+    this.pgBoss = pgBoss
+  }
+}
+
+class PgBossJob {
+  constructor(values) {
+    this.perform = () => { }
+    this.options = {}
+    this.startAfter = 0
+    this.jobName = 'unknown'
+    Object.assign(this, values)
+  }
+
+  delay(startAfter) {
+    return new PgBossJob({ ...this, startAfter })
+  }
+
+  async submit(payload, options) {
+    const jobId = await boss.send(this.jobName, payload, { ...this.options, startAfter: this.startAfter, ...options })
+    return new PgBossSubmittedJob(this.jobName, jobId, 'PgBoss', {
+      async cancel() { return boss.cancel(jobId) },
+      async resume() { return boss.resume(jobId) },
+      async details() { return boss.getJobById(jobId) }
+    })
+  }
+}
+
+export async function createJob(jobName, fn, options) {
+  await boss.work(jobName, fn)
```

Comment:
This just registers a worker callback. I can add a comment for it. So anytime we enqueue work for jobName, PgBoss calls fn. It can be called multiple times _if_ you want multiple workers listening for work. This should ideally only be used by the _job.js template. However, the name could be misleading for the side effect it has for sure! I was trying to abstract away that requirement, but it does probably need a different name 🤔

Reply:
Aha so it will register two workers for the same job name -> ok makes sense!